lebedigital/openbis/README.md

```diff
@@ -0,0 +1,140 @@
+        'property3': ['FLOAT', 'my_label2', 'my_description2']
+    }
+)
+```
+> You can find all the possible property types like 'VARCHAR' or 'FLOAT' [here](https://pypi.org/project/PyBIS/) under 'create property types'
+
+> You can also define the Sample Type in the Web-View of the Datastore
+### Now you can create the Experimental Step describing your research experiment. Start by declaring an object of the ExpStep class, which you want to upload to the Datastore
+
+#### An ExpStep Class instance looks like:
+
+``` python
+new_experiment = ExpStep(
+    name: str = 'exp_name', # name of the experiment
+    type: str = 'exp_type', # type of the experiment, has to be a defined sample type in the datastore
+    metadata: dict = exp_metadata, # metadata of your data, is defined by the sample type
+    space: str = 'exp_space', # space in which the experiment should be saved
+    project: str = 'exp_project', # project in which the experiment should be saved
+    collection: str = 'exp_space/exp_project/exp_collection', # collection in which the experiment should be saved
+    parents: list[str] = [exp_parents], # parents of the experiment
+    identifier: str = '', # identifier of the sample (SPACE/PROJECT/SAMPLE_CODE)
+    permId: str = '', # permID of the sample
+    sample_object=None, # Pybis object of the sample
+    datasets: list = None, # dataset objects saved in the sample
+    dataset_codes: list = None, # codes of the dataset objects saved in the sample
+)
+```
+
+> Note: You do not have to define all of those attributes, only the name, type, metadata and the location of the Sample in the Database (space, project, collection) are necessary
+
+### An example workflow which would upload your experimental step describing your concrete experiment could be 
+
+#### 1. Define the ExpStep as you see fit
+```python. 
+from lebedigital.openbis.expstep import ExpStep
+
+concrete_experiment = ExpStep()
+
+concrete_experiment.name = 'Concrete_Experiment_1'
+concrete_experiment.type = 'EXPERIMENTAL_STEP_CONCRETE_TEST'
+concrete_experiment.metadata = {'name': 'Concrete_Experiment_1', 'date': '17:10:2022'}
+
+# define the location in the datastore
+
+concrete_experiment.space = 'SPACE'
+concrete_experiment.project = 'PROJECT'
+concrete_experiment.collection = 'SPACE/PROJECT/COLLECTION'
+```
+
+#### 2. Upload the ExpStep to the Database
+```python
+concrete_experiment.upload_expstep(o) # o = connected Interbis instance
+```
+#### Good to know:
+* To upload a sample of a given type that type has to already exist in the datastore.
+* The name of the sample has to be the same as the `$name` parameter in the metadata dictionary in order to be able to find the sample efficiently again.
+* You can also load an existing sample into an ExpStep object using the `load_sample()` method of ExpStep.
+* Every time you want to upload some Sample to the Database a Type Checker will run in order to make sure your ExpStep has been defined properly.
```

Comment:
Note that only the types of the metadata dict are checked and not e.g. parent child connections or if dataset are given etc.